lib/ansible/modules/cloud/vmware/vmware_guest_network.py

```diff
@@ -432,6 +449,14 @@ def get_network_config_spec(self, vm_obj, network_list):
                                 nic_device.addressType = 'manual'
                                 nic_device.macAddress = network['manual_mac']
                                 self.change_detected = True
+                            if 'directpath_io' in network:
+                                if isinstance(nic_device, vim.vm.device.VirtualVmxnet3):
+                                    if nic_device.uptCompatibilityEnabled != network['directpath_io']:
+                                        nic_device.uptCompatibilityEnabled = network['directpath_io']
```

Comment:
This parameter "network['directpath_io']" can also be validated first in sanitize_network_params() function. Thanks.

Reply:
This is for reconfigure NIC, can this parameter be set for new created vmxnet3 adpater? Thanks.

Reply:
The only sanitization `network['directpath_io']` requires is to check if it is used with `vmxnet3` device type or not, I don't see a way to sanitize this in `sanitize_network_param()` function as `device_type` is not a mandatory parameter to identify the network adapter.
To check if the device type is `vmxnet3`, I will need the `nic` object which is not available in the function.

Are there any other validations that I am missing?

Reply:
Sorry for this late response.
You do not need to check device type in the sanitize function, just check the parameter 'directpath_io' is specified or not, if specified then check if it's bool, if not bool, I think the validation should return false, or come to this line:
`nic_device.uptCompatibilityEnabled = network['directpath_io']`, there might be error to set it to a not supported value. Thanks.